### PR TITLE
ipn/ipnserver: change Server to let LocalBackend be supplied async

### PR DESCRIPTION
**Pull request by bradfitz**
This is step 1 of de-special-casing of Windows and letting the LocalAPI HTTP server start serving immediately, even while the rest of the world (notably the Engine and its TUN device) are being created, which can take a few to dozens of seconds on Windows.

With this change, the ipnserver.New function changes to not take an Engine and to return immediately, not returning an error, and let its Run run immediately. If its ServeHTTP is called when it doesn't yet have a LocalBackend, it returns an error. A TODO in there shows where a future handler will serve status before an engine is available.

Future changes will:

* delete a bunch of tailscaled_windows.go code and use this new API
* add the ipnserver.Server ServerHTTP handler to await the engine being available
* use that handler in the Windows GUI client

Updates #6522

### CODE REVIEW

File: ipn/ipnserver/server.go [hunk 1]
```diff
@@ -11,69 +11,28 @@ import (
...
-
-	// LoginFlags specifies the LoginFlags to pass to the client.
-	LoginFlags controlclient.LoginFlags
-}
-
 // Server is an IPN backend and its set of 0 or more active localhost
 // TCP or unix socket connections talking to that backend.
 type Server struct {
-	b            *ipnlocal.LocalBackend
+	lb           syncs.AtomicValue[*ipnlocal.LocalBackend]
```
Comment (maisem) [inline ipn/ipnserver/server.go:35]: should this be a `atomic.Pointer`?

File: ipn/ipnserver/server.go [hunk 2]
```diff
@@ -304,43 +282,53 @@ func (s *Server) addActiveHTTPRequest(req *http.Request, ci *ipnauth.ConnIdentit
...
+	if _, ok := s.lb.LoadOk(); ok {
+		panic("already set")
+	}
+	// Yes, this LoadOk+Store is technically racy. But
+	// syncs.AtomicValue.CompareAndSwap doesn't work with with a nil source, as
+	// the zero value of the underlying atomic.Value is holding a nil interface
+	// and the typed CompareAndSwap wrapper uses a typed nil as the swap source.
+	// But this is fine being best effort. It's only meant to catch obvious
+	// start-up API mistakes. It's not called at runtime.
+	s.lb.Store(lb)
```
Comment (maisem) [inline ipn/ipnserver/server.go:313]: then you can simplify this bit too.

**Review by maisem — APPROVED** [review]

File: ipn/ipnserver/server.go (hunk 1 above)
Comment (bradfitz) [reply ipn/ipnserver/server.go:35]: sure, done.

File: ipn/ipnserver/server.go (hunk 2 above)
Comment (bradfitz) [reply ipn/ipnserver/server.go:313]: done

**Pull request merged**